### PR TITLE
Require all fields in signature, as always passed

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Require all fields in signature, as always passed

Before
----------------------------------------
All calls to the function pass all variables but they show up as optional

image

After
----------------------------------------
They are required

Technical Details
----------------------------------------
This is clearer & allows the addition of more required fields at the end

I did a universe search... it's not called from outside core code thankfully 

image

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Event/Form/Registration/Register.php [hunk 1]
```diff
@@ -588,15 +592,8 @@ public static function buildAmount(&$form, $required = TRUE, $discountId = NULL,
...
       // CRM-14492 Admin price fields should show up on event registration if user has 'administer CiviCRM' permissions
-      $adminFieldVisible = FALSE;
-      if (CRM_Core_Permission::check('administer CiviCRM data')) {
-        $adminFieldVisible = TRUE;
-      }
-
-      $hideAdminValues = TRUE;
-      if (CRM_Core_Permission::check('edit event participants')) {
-        $hideAdminValues = FALSE;
-      }
```
Comment (colemanw) [inline CRM/Event/Form/Registration/Register.php:599]: Sadly not the *most* overcomplicated method I've seen of assigning a boolean value to a variable.

Comment (eileenmcnaughton) [reply CRM/Event/Form/Registration/Register.php:599]: @colemanw there is competition!

Comment (colemanw) [reply CRM/Event/Form/Registration/Register.php:599]: It could have first assigned it to an undeclared class variable :)

Comment (eileenmcnaughton) [reply CRM/Event/Form/Registration/Register.php:599]: @colemanw there is that

**Pull request merged**